internal/controllers/addon/phase_observe_operatorresource.go

```diff
@@ -76,36 +79,44 @@ func (r *olmReconciler) observeOperatorResource(
 		return resultRetry, nil
 	}
 
+	// If CSV is present and is in succeeded phase we report
+	// the addon as available.
+	reportReadinessStatus(addon)
```

Comment:
We don't want to report this here itself. 

The addon-readiness is supposed to be reported only after all the subreconcilers have successfully reconciled.

If we report addon-readiness here itself, then addon-readiness wouldn't be affected by the failures of any other sub-reconciliations happening after the success of olm-subreconcilers.

Reply:
> If we report addon-readiness here itself, then addon-readiness wouldn't be affected by the failures of any other sub-reconciliations happening after the success of olm-subreconciler.

If there are errors in other sub-reconcilers this snippet would never run, since we return either the error or the requeueResult with retryAfter set.

Reply:
but the metricsFederation sub-reconciler runs after OLM reconciler. And in the future, there can be a different sub-reconciler which might run after the OLM sub-reconcilers.

And we might wanna reflect the readiness of the Addon only after ALL the sub-reconcilers execute without any `err`

Reply:
In that case, the metricsFederationReconciler should mark the addon is unavailable with the reason :`unreadyMetricsFederation`. (Like we do in other sub-reconciler)